File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
import java.util.*;

class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        
        for (int i = 0; i<numbers.length; i++) {
            long result = numbers[i];
            String target = Long.toBinaryString(numbers[i]);
            
            if (result % 2 == 0) {
                answer[i] = result + 1;
            } else {
                int idx = target.lastIndexOf("0");
                
                if (idx == -1) {
                    String tmp = "10" + target.substring(1, target.length());
                    answer[i] = Long.parseLong(tmp, 2);
                } else {
                    String tmp = target.substring(0, idx) + "10" + target.substring(idx + 2, target.length());
                    answer[i] = Long.parseLong(tmp, 2);
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.py
def solution(numbers):
    answer = []
    
    for n in numbers:
        binn = list('0' + bin(n)[2:])
        idx = ''.join(binn).rfind('0')
        binn[idx] = '1'
        
        if n % 2 == 1:
            binn[idx+1] = '0'
            
        answer.append(int(''.join(binn), 2))
    return answer